source/simulator/checkpoint_restart.cc

```diff
@@ -456,6 +457,11 @@ namespace aspect
       {
         AssertThrow(false, ExcMessage("Cannot open snapshot mesh file or read the triangulation stored there."));
       }
+
+    // if using a cached mapping, update the cache with the new triangulation
+    if (MappingQCache<dim> *map = dynamic_cast<MappingQCache<dim>*>(&(*mapping)))
+      map->initialize(triangulation,MappingQGeneric<dim>(4));
```

Comment:
Is there an easy way to query the 4 from the cache? I would like to avoid having duplicated logic like this.

Reply:
Not at the moment, because we do not actually store anything related to the mapping. It is constructed upon start/ after restart and the 4 is just our currently hard-coded degree. But I see why you are concerned. I see two options: Storing the degree of the mapping in a Simulator member variable that gets serialized into the checkpoint, or making the mapping degree an input parameter. The first guarantees that a restart uses the same degree, the second does not (but at least removes the duplicated copies in the code if we use parameters.mapping_degree in both the constructor and here). One complication is that this variable really only makes sense for spherical shells at the moment, boxes always use first order mappings (either MappingCartesian or MappingQ1Eulerian).

Reply:
Exactly. The parameter with default 4 might be the best option. Anyways, I will merge and leave it up to you if you want to make that change later.